src/actions/blockchain.ts

```diff
@@ -201,6 +219,111 @@ export const initDutchX = () => async (dispatch: Dispatch<any>, getState: () =>
   }
 }
 
+export const setApprovedTokensAndAvailableAuctions = (tokenList: DefaultTokenList) => async (dispatch: Dispatch<any>) => {
+  console.log('​exportsetApprovedTokensAndAvailableAuctions -> tokenList', tokenList)
+  const [approvedTokenAddresses, availableAuctions] = await Promise.all([
+    getApprovedTokensFromAllTokens(tokenList),
+    getAvailableAuctionsFromAllTokens(tokenList),
+  ])
+
+  console.log(`
+    APPROVED TOKEN ADDRESSES: ${JSON.stringify(approvedTokenAddresses, undefined, 2)}
+    AVAILABLE AUCTIONS: ${JSON.stringify(availableAuctions, undefined, 2)}
+  `)
+
+  dispatch(setApprovedTokens(approvedTokenAddresses))
+  dispatch(setAvailableAuctions(availableAuctions))
+}
+
+export const getTokenList = (network?: number | string) => async (dispatch: Dispatch<any>, getState: () => State) => {
+
+  let [defaultTokens, customTokens, customListHash] = await Promise.all<DefaultTokens, DefaultTokens['elements'], string>([
+    localForage.getItem('defaultTokens'),
+    localForage.getItem('customTokens'),
+    localForage.getItem('customListHash'),
+  ])
+  const { ipfsFetchFromHash } = await promisedIPFS
+
+  // when switching Networks, NetworkListeners in events.js should delete localForage tokenList
+  // meaning this would be FALSE on network change and app reset
+  const isDefaultTokensAvailable = !!(defaultTokens)
+
+  if (!isDefaultTokensAvailable) {
+    network = network || 'NONE'
+
+    console.log('Current Network =', network)
+
+    // grab tokens from IPFSHash or api/apiTesting depending on NODE_ENV
+    /* if (process.env.NODE_ENV === 'development') {
+      defaultTokens = await ipfsFetchFromHash(IPFS_TOKENS_HASH) as DefaultTokens
+    } else {
+      // TODO: change for prod
+      defaultTokens = await ipfsFetchFromHash(IPFS_TOKENS_HASH) as DefaultTokens
+    } */
+
+    switch (network) {
+      case 'RINKEBY' || '4' || ETHEREUM_NETWORKS.RINKEBY:
```

Comment:
This isn't how `switch..case` works. Here it'll compare `network === 'RINKEBY' || '4' || ETHEREUM_NETWORKS.RINKEBY`, which is the same as `network === 'RINKEBY'`.

You want a fallthrough:
```
case 'RINKEBY':
case '4':
case ETHEREUM_NETWORKS.RINKEBY:
  <code>
```